test/integration_test/applicationbackup_test.go

```diff
@@ -55,36 +78,56 @@ func triggerBackupRestoreTest(
 	}
 
 	// Backup application
-	err = schedulerDriver.AddTasks(ctxs[0],
-		scheduler.ScheduleOptions{AppKeys: appBackupKey})
+	if !backupAllAppsExpected && backupSuccessExpected {
+		err = schedulerDriver.AddTasks(postRestoreCtx, scheduler.ScheduleOptions{AppKeys: appBackupKey})
+	} else {
+		err = schedulerDriver.AddTasks(ctxs[0], scheduler.ScheduleOptions{AppKeys: appBackupKey})
+	}
 	require.NoError(t, err, "Error backing-up apps")
+
 	if backupSuccessExpected {
-		err = schedulerDriver.WaitForRunning(ctxs[0], defaultWaitTimeout, defaultWaitInterval)
+		if backupAllAppsExpected {
+			err = schedulerDriver.WaitForRunning(ctxs[0], defaultWaitTimeout, defaultWaitInterval)
+		} else {
+			err = schedulerDriver.WaitForRunning(postRestoreCtx, defaultWaitTimeout, defaultWaitInterval)
+		}
 		require.NoError(t, err, "Error waiting for back-up to complete.")
-		logrus.Infof("Backup completed. Starting Restore.")
+		logrus.Infof("Backup completed.")
 
+		// Destroy apps so that restore can be verified
+		destroyAndWait(t, []*scheduler.Context{preRestoreCtx})
```

Comment:
If you destroy these then the `Delete` Reclaim policy won't be tested anymore, right?

Reply:
Added check.